src/app/slice/postSlice.js

```diff
@@ -0,0 +1,124 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+import { getAllPosts, getAllUsers, getPostById, getPostByUsername, getUserById } from "../../Helpers/Services/actions"; 
+// import { updateUser } from "../Auth/authSlice";
+
+const initialState = {
+  status:'idle', // 'idle' | 'loading' | 'suceeded' | 'failed'
+  error:'null'
+};
+
+export const getPosts = createAsyncThunk(
+    "post/getPosts",
+    async (_, thunkAPI) => {
+      try {
+          const response = await getAllPosts();
+          return response.data.posts;
+      } catch (error) {
+        return thunkAPI.rejectWithValue(error);
+      }
+    }
+  );
+
+  //getPostById
+  export const getPostDetailsByID = createAsyncThunk(
+    "post/getPostDetailsByID",
+    async ( postId , thunkAPI) => {
+      try {
+          const response = await getPostById(postId);
+          return response.data;
+      } catch (error) {
+        return thunkAPI.rejectWithValue(error);
+      }
+    }
+  );
+
+  //getPostByUsername
+  export const getPostBy_username = createAsyncThunk(
+    "post/getPostBy_username",
+    async ( username , thunkAPI) => {
+      try {
+          const response = await getPostByUsername(username);
+          return response.data;
+      } catch (error) {
+        return thunkAPI.rejectWithValue(error);
+      }
+    }
+  );
+
+  const postSlice = createSlice({
+    name: "post",
+    initialState,
+    reducers: {},
+    extraReducers: {
+        [getPosts.pending]: (state, action) => {
+        state.posts = {
```

Comment:
this state.posts is not in initial state

Reply:
ok, managed using state.posts